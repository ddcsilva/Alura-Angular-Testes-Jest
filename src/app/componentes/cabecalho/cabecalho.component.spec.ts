import { ComponentFixture, TestBed } from '@angular/core/testing';
import { CabecalhoComponent } from './cabecalho.component';

describe('CabecalhoComponent', () => {
  let component: CabecalhoComponent;
  let fixture: ComponentFixture<CabecalhoComponent>;

  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [CabecalhoComponent],
    });

    fixture = TestBed.createComponent(CabecalhoComponent);
    component = fixture.componentInstance;
  });

  it('deveria ser criado', () => {
    // Assert
    expect(component).toBeTruthy();
  });

  it('deveria definir as propriedades alt e src', () => {
    // Assert
    expect(component.alt).toBeDefined();
    expect(component.src).toBeDefined();
  });

  it('deveria renderizar o conteÃºdo baseado nas propriedades alt e src', () => {
    // Arrange
    component.alt = 'Logo do Sistema';
    component.src = 'assets/logo.png';

    // Act & Assert
    expect(component).toMatchSnapshot();
  });
});
