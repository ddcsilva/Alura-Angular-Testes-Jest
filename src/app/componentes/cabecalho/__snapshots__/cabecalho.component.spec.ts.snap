// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CabecalhoComponent deveria renderizar o conte√∫do baseado nas propriedades alt e src 1`] = `
CabecalhoComponent {
  "__ngContext__": 4,
  "alt": "Logo do Sistema",
  "src": "assets/logo.png",
}
`;
